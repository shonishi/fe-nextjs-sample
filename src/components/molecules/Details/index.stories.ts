import type { Meta, StoryObj } from '@storybook/react';
import Details from '.';

const meta: Meta<typeof Details> = {
  component: Details,
};

export default meta;
type Story = StoryObj<typeof Details>;

export const Normal: Story = {
  args: {
    details:
      'かわいいあひるちゃんの5羽セット。シリコン製のため小傷が入りにくく、衛生面も安心です。音を出すにはある程度の力で握る必要がありますので、楽しみながらお子様の筋力を強化することができます。各種サイズをご用意しておりますので、お子様の手の大きさに合う物をお選びいただけます。',
  },
};

export const Empty: Story = {
  args: {
    details: '',
  },
};
