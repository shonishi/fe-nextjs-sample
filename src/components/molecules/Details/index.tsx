interface Props {
  details: string;
}

export default function Details({ details }: Props) {
  return (
    <div className="mt-10">
      <h2 className="text-sm font-medium text-gray-900">Details</h2>
      <div className="mt-4 space-y-6">
        <p className="text-sm text-gray-600">{details}</p>
      </div>
    </div>
  );
}
