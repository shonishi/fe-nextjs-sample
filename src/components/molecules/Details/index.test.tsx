import Details from '.';
import renderer from 'react-test-renderer';

describe('Details', () => {
  it('detailsに文字列が設定されている場合、それを元に表示されること', async () => {
    const tree = renderer
      .create(
        <Details
          {...{
            details: 'あひるのフィギュアです。お風呂に浮かべて楽しいひと時。',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('detailsが空の場合、空で表示されること', async () => {
    const tree = renderer
      .create(
        <Details
          {...{
            details: '',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
