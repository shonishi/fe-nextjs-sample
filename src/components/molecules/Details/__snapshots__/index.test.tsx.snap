// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details detailsが空の場合、空で表示されること 1`] = `
<div
  className="mt-10"
>
  <h2
    className="text-sm font-medium text-gray-900"
  >
    Details
  </h2>
  <div
    className="mt-4 space-y-6"
  >
    <p
      className="text-sm text-gray-600"
    />
  </div>
</div>
`;

exports[`Details detailsに文字列が設定されている場合、それを元に表示されること 1`] = `
<div
  className="mt-10"
>
  <h2
    className="text-sm font-medium text-gray-900"
  >
    Details
  </h2>
  <div
    className="mt-4 space-y-6"
  >
    <p
      className="text-sm text-gray-600"
    >
      あひるのフィギュアです。お風呂に浮かべて楽しいひと時。
    </p>
  </div>
</div>
`;
